blog/content/post/2022-12-16-trenchboot-aem-for-qubesos.md

```diff
@@ -0,0 +1,358 @@
+be done was the Multiboot2 protocol support in GRUB to be able to TXT launch a
+Xen Hypervisor. The patches have been prepared for the respective Qubes GRUB
+[package](https://github.com/3mdeb/qubes-grub2/pull/2).
+
+## Xen modifications
+
+![](/img/xen_project_logo.png)
+
+Analogically to GRUB, Xen had to take over some responsibilities from tboot.
+Due to the Intel TXT requirements for the boot process, a new entry point had
+to be developed to which SINIT ACM will return control. The new entry point was
+responsible for saving information that a TXT launch happened and cleaning up
+the processor state so that the booting of the Xen kernel could continue with
+the standard Multiboot2 path. Among others, if Xen detected TXT launch, it had
+to perform the special processor cores wakeup process (which has been rewritten
+from TrenchBoot Linux patches to Xen native code) and measure external
+components before using them (that is the Xen parameters, Dom0 Linux kernel,
+initrd and Dom0 parameters). Xen also had to reserve the memory regions used by
+Intel TXT, as when tboot was used. The relevant source code for the respective
+Qubes Xen package is available [here](https://github.com/3mdeb/qubes-vmm-xen/pull/1).
+
+# Installation and verification of TrenchBoot AEM on Qubes OS
+
+For a seamless deployment and installation of TrenchBoot AEM, the modifications
+have been converted to patches which are applied to projects' sources during
+Qubes OS components compilation. Those patches have been presented earlier with
+links to Pull Requests. It allows building ready-to-use RPM packages that can
+be installed directly on an installed Qubes OS system. The pre-built packages
+can be downloaded from [here](https://3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_poc/).
+The packages have been covered with SHA512 sums signed with 3mdeb's
+`Qubes OS TrenchBoot AEM open-source software release 0.x signing key`
+available on [3mdeb-secpack repository](https://github.com/3mdeb/3mdeb-secpack/blob/master/open-source-software/qubes-os-trenchboot-aem-open-source-software-release-0.x-signing-key.asc).
+To verify the RPM packages, fetch the key with the following command:
+`gpg --fetch https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/open-source-software/qubes-os-trenchboot-aem-open-source-software-release-0.x-signing-key.asc`
+and then for each package, please run:
+
+```txt
+gpg --verify <rpm_package-name>.sha512.sig <rpm_package-name>.sha512
+sha512sum -c <rpm_package-name>.sha512
+```
+
+Check if GPG returns a good signature and if yes, check if the RPM checksum
+matches. All files must be in the same directory for the procedure to work.
+
+Note, in order to use the TrenchBoot AEM for Qubes OS, you have to own a
+TXT-capable platform with TXT-enabled firmware offering legacy boot. You may
+find such platform and firmware in the [Dasharo with Intel TXT support](https://blog.3mdeb.com/2022/2022-03-17-optiplex-txt/)
+post.
```

Comment:
We link to OptiPlex platform, but I think we should mention that hardware here in plain text, also link to shop would be appreciated. D-RTM capable platform maybe interesting to community members.

Reply:
Resolved in: https://github.com/3mdeb/news-and-ideas/pull/503/commits/51d8b96d397437da8715495891dbe550493ad3f4